src/Angor/Client/Pages/Signatures.razor

```diff
@@ -161,6 +197,49 @@
         }
     }
 
+    private async Task RefreshSignaturesInternal()
+    {
+        if (passwordComponent.HasPassword())
+        {
+            await RefreshSignatures();
+        }
+        else
+        {
+            passwordComponent.ShowPassword(async () =>
+            {
+                await RefreshSignatures();
+            });
+        }
+    }
+
+    private async Task RefreshSignatures()
+    {
+        
+        
+        refreshButtonSpinner = true;
+        StateHasChanged();
+        
+
+        try
+        {
+            await FetchPendingSignatures(FounderProject);
+            await FetchSignatures();
+            await Task.Delay(1000);
+        }
+        catch (Exception e)
+        {
+            Logger.LogDebug($"Error fetching signatures: {e.Message}");
+        }
+        finally
+        {
+            
+            refreshButtonSpinner = false;
+            StateHasChanged();
+        }
+    }
+
+
+
     protected async Task FetchSignatures()
     {
         Logger.LogDebug("handled = {Count}, total = {SignaturesRequestsCount}", signaturesRequests.Count(x => x.AmountToInvest.HasValue), signaturesRequests.Count);
```

Comment:
Why not use string interpolation?

Reply:
For some reason structured logging is more readable for me, changed it